### DISCUSSION TITLE
How to bind feature file and step definition in Visual studio code for Python Behave?

### DISCUSSION THREAD

**Discussion by Ashwinigirish** [behave/behave#1104] [Q&A]
Hello every one,

I am unable to bind feature file and step definition in visual studio code for Python behave.
When I right click on step definition. It says "no definition found for 'given'"
Please find the attached screenshot.
![image](https://user-images.githubusercontent.com/131980163/234879648-eae2e959-d080-4023-9ced-cd84e4df1ea0.png)
Also, when I try to run through terminal it says "No steps directory in the specified path".
Is any problem with folder structure ?
Please help me how to solve these two issues.

**Comment by bittner** [comment]
You may want to take a look at the IDE extensions available in the Mircosoft marketplace and their related tutorials, e.g. https://duckduckgo.com/?q=vscode+behave

Other than that, the [official Behave documentation](https://behave.readthedocs.io/en/stable/tutorial.html) describes step by step how to get your initial setup working. The directory structure seems to be alright according to your screenshot, maybe you're in the wrong directory when running `behave`, though.

As a side-note: In general, please paste the entire input and error output when reporting a problem. That makes it easier to analyse and understand the problem, to be able to help you.

**Comment by jenisys** [comment]
@Ashwinigirish 
* First, you need to describe which VSCode extension you are using (and the problem and solution will normally be there)
* Normally, feature-file(s) are contained in a `features/` directory (as a convention, not a requirement)
* When you click on the `@given` Python decorator, you are already at the step-implementation. 
   A step can be used in multiple features and scenarios.
* Normally, you click on the step in the `feature-file` and are navigated to the step-implementation function (where you already are in your screenshot)
